dot/core/service.go

```diff
@@ -140,7 +140,11 @@ func (s *Service) HandleBlockProduced(block *types.Block, state *rtstorage.TrieS
 
 	digest := types.NewDigest()
 	for i := range block.Header.Digest.Types {
-		err := digest.Add(block.Header.Digest.Types[i].Value())
+		digestValue, err := block.Header.Digest.Types[i].Value()
+		if err != nil {
+			return err
```

Comment:
nit bunble some context and the index in the error context (only accessible at this level):

```suggestion
			return fmt.Errorf("getting value of digest type at index %d: %w", i, err)
```